contracts/protocol/libraries/logic/ReserveLogic.sol

```diff
@@ -178,6 +178,23 @@ library ReserveLogic {
     reserve.interestRateStrategyAddress = interestRateStrategyAddress;
   }
 
+  /**
+   * @dev drop a reserve
+   * @param reserve The reserve object
+   **/
+  function dropReserve(DataTypes.ReserveData storage reserve) external {
+    require(IERC20(reserve.aTokenAddress).totalSupply() == 0, Errors.RL_ATOKEN_SUPPLY_NOT_NULL);
+    require(
+      IERC20(reserve.stableDebtTokenAddress).totalSupply() == 0,
+      Errors.RL_STABLE_DEBT_NOT_NULL
+    );
+    require(
+      IERC20(reserve.variableDebtTokenAddress).totalSupply() == 0,
+      Errors.RL_VARIABLE_DEBT_SUPPLY_NOT_NULL
+    );
+    reserve.id = type(uint8).max;
```

Comment:
i think reserve.id can be set to 0, that is the default for reserves not configured yet - ultimately deleting the reserveData object should do the trick